Lib/test/test_asyncio/test_selector_thread.py

```diff
@@ -0,0 +1,123 @@
+import asyncio
+import select
+import socket
+import time
+import unittest
+from asyncio._selector_thread import SelectorThread
+from unittest import mock
+
+
+class SelectorThreadTest((unittest.IsolatedAsyncioTestCase)):
+    async def asyncSetUp(self):
+        self._sockets = []
+        self.selector_thread = SelectorThread(asyncio.get_running_loop())
+
+    def socketpair(self):
+        pair = socket.socketpair()
+        self._sockets.extend(pair)
+        return pair
+
+    async def asyncTearDown(self):
+        self.selector_thread.close()
+        for s in self._sockets:
+            s.close()
+
+    async def test_slow_reader(self):
+        a, b = self.socketpair()
+        first_recv = asyncio.Future()
+
+        def recv():
+            msg = b.recv(100)
+            if not first_recv.done():
+                first_recv.set_result(msg)
+
+        mock_recv = mock.MagicMock(wraps=recv)
+        # make sure select is only called once when
+        # event loop thread is slow to consume events
+        a.sendall(b"msg")
+        with mock.patch("select.select", wraps=select.select) as mock_select:
+            self.selector_thread.add_reader(b, mock_recv)
+            # ready event, but main event loop is blocked for some time
+            time.sleep(0.1)
+            recvd = await asyncio.wait_for(first_recv, timeout=10)
```

Comment:
```suggestion
            recvd = await asyncio.wait_for(first_recv, timeout=support.SHORT_TIMEOUT)
```

You need to add `from test import support` at the top.